tethysext/atcore/controllers/resource_workflows/results_views/report_workflow_results_view.py

```diff
@@ -0,0 +1,218 @@
+    SpatialWorkflowResult
+
+from tethys_sdk.gizmos import DataTableView
+from collections import OrderedDict
+
+
+log = logging.getLogger(__name__)
+
+
+class ReportWorkflowResultsView(MapWorkflowView, WorkflowResultsView):
+    """
+    Report Result View controller.
+    """
+    template_name = 'atcore/resource_workflows/report_workflow_results_view.html'
+    valid_result_classes = [ReportWorkflowResult]
+
+    previous_steps_selectable = True
+
+    def process_step_options(self, request, session, context, resource, current_step, previous_step, next_step):
+        """
+        Hook for processing step options (i.e.: modify map or context based on step options).
+
+        Args:
+            request(HttpRequest): The request.
+            session(sqlalchemy.orm.Session): Session bound to the steps.
+            context(dict): Context object for the map view template.
+            resource(Resource): the resource for this request.
+            current_step(ResourceWorkflowStep): The current step to be rendered.
+            previous_step(ResourceWorkflowStep): The previous step.
+            next_step(ResourceWorkflowStep): The next step.
+        """
+        # Turn off feature selection on model layers
+        map_view = context['map_view']
+        self.set_feature_selection(map_view=map_view, enabled=False)
+
+        # Can run workflows if not readonly
+        can_run_workflows = not self.is_read_only(request, current_step)
+
+        # get tabular data if any
+        tabular_data = current_step.workflow.get_tabular_data_for_previous_steps(current_step)
+        has_tabular_data = len(tabular_data) > 0
+
+        # Generate MVLayers for spatial data
+        # Get managers
+
+        _, map_manager = self.get_managers(
+            request=request,
+            resource=resource,
+        )
+
+        # Get DatasetWorkflowResult
+        results = list()
+        for result in current_step.results:
+            if isinstance(result, DatasetWorkflowResult):
+                for ds in result.datasets:
+                    # Check if the export options is there
+                    data_table = DataTableView(
+                        column_names=ds['dataset'].columns,
+                        rows=[list(record.values()) for record in ds['dataset'].to_dict(orient='records',
+                                                                                        into=OrderedDict)],
+                        searching=False,
+                        paging=False,
+                        info=False
+                    )
+                    ds.update({'data_table': data_table})
+                    ds.update({'data_description': result.description})
+                    results.append({'dataset': ds})
+            elif isinstance(result, PlotWorkflowResult):
+                results.append({'plot': [result.name, result.description, result.get_plot_object()]})
```

Comment:
Use a dictionary instead of a list so that it is easier to follow when used in the template. For example:

```python
[result.name, result.description, result.get_plot_object()]
```
ends ups being used like this in the template:

```html
<h3>{{ v.0 }}</h3>
 {% if v.1 %}
     <h4>Description: {{ v.1 }}</h4>
 {% endif %}
<div id="resultPlotDiv{{ forloop.counter }}" style="width:800px;">
     {% gizmo v.2 %}
</div>
```

If you change it to a dictionary like:

```python
{'name': result.name, 'description': result.description, 'gizmo': result.get_plot_object()}
```

Then the template is easier to read:

```html
<h3>{{ v.name }}</h3>
 {% if v.description %}
     <h4>Description: {{ v.description }}</h4>
 {% endif %}
<div id="resultPlotDiv{{ forloop.counter }}" style="width:800px;">
     {% gizmo v.gizmo %}
</div>
```